docs/source/fx.rst

```diff
@@ -10,7 +10,154 @@ Overview
+a new graph (which may simply be a copy of the input), and then return
+an ``nn.Module``. You should think of the ``nn.Module`` that your FX
+transform returns as identical to a regular ``nn.Module``- you can pass
+it to another FX transform, you can pass it to Torchscript, or you can
+run it. Ensuring that the inputs and outputs of your FX transform are a
+``nn.Module`` will ensure composability.
+
+Given that you’ve passed in a ``nn.Module`` that has been traced into a
+graph, there are now 2 primary approaches you can take to building a new
+graph.
+
+Graph Manipulation
+------------------
+
+One approach to building this new graph is to simply transform your old
+one. To aid in this, we can simply take the graph we obtain from
+symbolic tracing and modify it. For example, let’s say we desired to
+replace ``torch.add`` with ``torch.mul``.
+
+::
+
+    # Sample module
+    class M(torch.nn.Module):
+        def forward(self, x, y):
+            return torch.add(x, y)
+
+    # Symbolically trace an instance of the module
+    traced = symbolic_trace(original_module)
+
+    for node in traced.graph.nodes:
+        if n.target == torch.add:
+            n.target = torch.mul
+
+    traced.recompile()
+
+We can also do more involved graph rewrites, such as appending a ReLU
+after a node. To aid in these transformations, FX also has utility
+functions for transforming the graph. You can find more .
+
+::
+
+    with traced.graph.inserting_after(node):
+        new_node = traced.graph.call_function(torch.relu, args=(node,))
+        node.replace_all_uses_with(new_node)
+
+This approach is also a good fit for graph optimizations - such as
+`conv/batch norm
+fusion! <https://github.com/pytorch/pytorch/blob/ec86cec20a8a2312a2295d7bc8be6e88256a2de4/torch/fx/experimental/fuser.py>`__
+
+For simple transformations that are simply substitutions, you can also
+make use of the `subgraph rewriter. <https://github.com/pytorch/pytorch/blob/master/torch/fx/subgraph_rewriter.py>`__
+
+In general, writing your transformation through graph manipulation is a
+good fit if you need to make a few small changes or if you need a global
+view of your graph. However, if you need to entirely rewrite your graph, you may want to look at
+constructing your graph with Proxies (i.e: retracing).
+
+Examples
+~~~~~~~~
+
+-  `Replace one
+   op <https://github.com/pytorch/pytorch/blob/master/torch/fx/examples/replace_op.py>`__
+-  `Conv/Batch Norm
+   fusion <https://github.com/pytorch/pytorch/blob/master/torch/fx/experimental/fuser.py>`__
+-  `Quantization <https://github.com/pytorch/pytorch/tree/master/torch/quantization/fx>`__
+
+Proxy/Retracing
+---------------
+
+Although most transformations can be implemented as graph
+transformations, transformations that involve large amounts of rewriting
+are often easier represented through retracing. For example, let’s
+imagine that we wanted to write a decomposition pass that decomposed
```

Comment:
You might just say "a pass that decomposed PyTorch functions" so you don't have the repetition, but that's just personal preference